packages/calcite-components/src/components/color-picker/utils.ts

```diff
@@ -261,17 +261,26 @@ export function toNonAlphaMode(mode: SupportedMode): SupportedMode {
   return nonAlphaMode;
 }
 
-export function getSliderWidth(activeDimensions: (typeof DIMENSIONS)[Scale], hasAlpha: boolean): number {
-  const {
-    slider: { width },
-    preview,
-  } = activeDimensions;
+const borderWidthInPx = 1;
+const inlineSizeBorderTotalWidth = borderWidthInPx * 2;
 
-  if (hasAlpha) {
-    return width;
-  }
+export function getSliderWidth(
+  availableWidth: number,
+  activeStaticDimensions: (typeof STATIC_DIMENSIONS)[Scale],
+  hasAlpha: boolean,
+): number {
+  const previewWidth = hasAlpha ? STATIC_DIMENSIONS["l"].preview.size : activeStaticDimensions.preview.size;
+  const effectiveWidth = availableWidth - inlineSizeBorderTotalWidth;
+
+  return Math.max(effectiveWidth - activeStaticDimensions.gap * 3 - previewWidth, 0);
```

Comment:
What is the `3` for? Just curious.

Reply:
It's the spacing between elements around the slider.

<img width="324" alt="Screenshot 2025-01-28 at 1 18 53 PM" src="https://github.com/user-attachments/assets/59ea37a7-d52e-46f2-ad11-257a83747b09" />

I should avoid the magic number though. WDYT about this?

```tsx
const inlineSpaceAroundElements = activeStaticDimensions.gap * 3;
```

Reply:
Ah, yeah i was just wondering where the magic number came from :)